ext/ddtrace.stub.php

```diff
@@ -0,0 +1,496 @@
+     * stack if no stack is given.
+     * @param SpanData|SpanStack|null $newStack Stack to switch to. If 'null' is given, switches to the parent of the
+     * active stack. If a SpanData object is given, it will switch to the stack of the latter.
+     * @return SpanStack|null The newly active stack, or 'null' if there wasn't an active stack and no specific stack
+     * was given as an argument to the function call
+     */
+    function switch_stack(SpanData|SpanStack|null $newStack = null): null|SpanStack {}
+
+    /**
+     * Set the priority sampling level
+     * @param int $priority The priority level to be set to
+     * @param bool|null $global If set to 'true' and if there is no active stack (or the active stack doesn't have a
+     * root span), then the default priority sampling will be set to the provided priority level. Otherwise, the root's
+     * priority sampling level will be updated with the new value.
+     * @return void
+     */
+    function set_priority_sampling(int $priority, ?bool $global = false): void {}
+
+    /**
+     * Get the priority sampling level
+     * @param bool|null $global If set to 'true' and if there is no active stack (or the active stack doesn't have a
+     * root span), then the default priority sampling will be returned, else it will be fetched from the root.
+     * @return int The priority sampling level
+     */
+    function get_priority_sampling(?bool $global = false): int {}
+
+    /**
+     * Sanitize an exception
+     * @internal
+     * @param \Exception $exception
+     * @return string
+     */
+    function get_sanitized_exception_trace(\Exception $exception): string {}
+
+    /**
+     * Update datadog headers for distributed tracing for new spans. Also applies this information to the current trace,
+     * if there is one, as well as the future ones if it isn't overwritten
+     * @param callable(string):mixed $func Given a header name for distributed tracing, return the value it should
+     * be updated to
+     * @return void
+     */
+    function consume_distributed_tracing_headers(callable $func): void {}
+
+    /**
+     * Get information on the key-value pairs of the datadog headers for distributed tracing
+     * @return array{x-datadog-sampling-priority: string,
+     *               x-datadog-origin: string,
+     *               x-datadog-trace-id: string,
+     *               x-datadog-parent-id: string,
+     *               traceparent: string,
+     *               tracestate: string
+     *          }
+     */
+    function generate_distributed_tracing_headers(): array {}
+
+    /**
+     * Check for active exception
+     * @return \Exception|null The active exception if there is one, else 'null'
+     */
+    function find_active_exception(): \Exception|null {}
+
+    /**
+     * Retrieve IPs from the given array if valid headers are found, and return them in
+     * a metadata formatting
+     * @param array{string: string} $headers
```

Comment:
```suggestion
     * @param string[] $headers
```
{string: string} means an array of format `["string" => "some string value"]`